src/erl.rs

```diff
@@ -1321,31 +1448,52 @@ fn incrementing_args_list(arity: usize) -> String {
         .collect()
 }
 
-fn external_fun<'a>(name: &str, module: &str, fun: &str, arity: usize) -> Document<'a> {
-    let chars: String = incrementing_args_list(arity);
+fn external_fun<'a>(
+    name: &'a str,
+    module: &'a str,
+    fun: &'a str,
+    args: &'a [TypedExternalFnArg],
+    return_type: &'a Type,
+) -> Document<'a> {
+    let chars: String = incrementing_args_list(args.len());
 
-    atom(name.to_string())
-        .append("(")
-        .append(Document::String(chars.clone()))
-        .append(") ->")
-        .append(line())
-        .append(atom(module.to_string()))
-        .append(":")
-        .append(atom(fun.to_string()))
-        .append("(")
-        .append(Document::String(chars))
-        .append(").")
-        .nest(INDENT)
+    let args_spec = args.iter().map(|a| a.typ.to_erlang_type_spec());
+    let return_spec = return_type.to_erlang_type_spec();
+    let spec = fun_spec(name, args_spec, return_spec);
+
+    spec.append(atom(name.to_string())).append(
+        Document::String(format!("({}) ->", chars))
+            .append(line())
+            .append(atom(module.to_string()))
+            .append(":")
+            .append(atom(fun.to_string()))
+            .append(Document::String(format!("({}).", chars)))
+            .nest(INDENT)
+            .group(),
+    )
 }
 
-fn variable_name(name: String) -> String {
+fn variable_name(name: &str) -> String {
     let mut c = name.chars();
     match c.next() {
         None => String::new(),
         Some(f) => f.to_uppercase().chain(c).collect(),
     }
 }
 
+// When rendering a type variable to an erlang type spec we need all type variables with the
+// same id to end up with the same name in the generated erlang.
+// This function maps ids to A-Z then AA-AZ, but then AAA - AAZ, rather than BA-BZ etc.
+// This effectively gives us 52 type vars per file until they start looking weird.
+// This is because to properly cycle through the alphabet we have to convert the usize to base 26.
+// If we need more we can always modulus one more time and that'll give us 26*26 good looking vars.
+// Alternatively we could have the id itself after a character like, `A0` and `A24`.
```

Comment:
Thanks for this excellent comment. I think I would like to raise the number of pretty looking type vars past 52